content/v2.0/query-data/flux/moving-average.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Calculate the moving average
+seotitle: Calculate the moving average in Flux
+list_title: Moving Average
+description: >
+  Use the [`movingAverage()`](/v2.0/reference/flux/stdlib/built-in/transformations/aggregates/movingaverage/)
+  or [`timedMovingAverage()`](/v2.0/reference/flux/stdlib/built-in/transformations/aggregates/timedmovingaverage/)
+  functions to return the moving average of data.
+weight: 210
+menu:
+  v2_0:
+    parent: Query with Flux
+    name: Moving Average
+v2.0/tags: [query, moving average]
+list_code_example: |
+  ```js
+  data
+    |> movingAverage(n: 5)
+
+  // OR
+
+  data
+    |> timedMovingAverage(every: 5m, period: 10m)
+  ```
+---
+
+Use the [`movingAverage()`](/v2.0/reference/flux/stdlib/built-in/transformations/aggregates/movingaverage/)
+or [`timedMovingAverage()`](/v2.0/reference/flux/stdlib/built-in/transformations/aggregates/timedmovingaverage/)
+functions to return the moving average of data.
+
+```js
+data
+  |> movingAverage(n: 5)
+
+// OR
+
+data
+  |> timedMovingAverage(every: 5m, period: 10m)
+```
+
+#### movingAverage()
+For each row in a table, `movingAverage()` returns the average of the current
+value and the previous `n-1` values.
+
+#### timedMovingAverage()
+For each row in a table, `timedMovingAverage()` returns the the average of the
+current value and all values in the previous `period` (duration).
+It returns moving averages at a frequency defined by the `every` parameter.
+
+**Given the following input:**
+
+| _time                | _value |
+|:-----                | ------:|
+| 2020-01-01T00:01:00Z | 1.0    |
+| 2020-01-01T00:02:00Z | 1.2    |
+| 2020-01-01T00:03:00Z | 1.8    |
+| 2020-01-01T00:04:00Z | 0.9    |
+| 2020-01-01T00:05:00Z | 1.4    |
+| 2020-01-01T00:06:00Z | 2.0    |
+
+**The following functions would return:**
+
+---
+
+{{< flex >}}
+{{% flex-content %}}
+**Function:**  
+
+```js
+// ...
+  |> movingAverage(n: 3)
+```
+{{% /flex-content %}}
+{{% flex-content %}}
+**Output:**  
+
+| _time                | _value |
+|:-----                | ------:|
+| 2020-01-01T00:03:00Z | 1.33   |
+| 2020-01-01T00:04:00Z | 1.30   |
+| 2020-01-01T00:05:00Z | 1.06   |
+| 2020-01-01T00:06:00Z | 1.43   |
+{{% /flex-content %}}
+{{< /flex >}}
+
+---
+
+{{< flex >}}
+{{% flex-content %}}
+**Function:**  
+
+```js
+// ...
+  |> timedMovingAverage(
+    every: 2m,
+    period: 4m
+  )
+```
+{{% /flex-content %}}
+{{% flex-content %}}
+**Output:**  
+
+| _time                | _value |
+|:-----                | ------:|
+| 2020-01-01T00:04:00Z | 1.23   |
```

Comment:
Where current time is n, the timed moving average is calculated for each n - period, increasing by the specified `every` interval until all input rows have been averaged. For example:

timed moving average = 
[(value at n) + (value at each point between n- period)]/number of points (rows),
[(value at n + every) + (value at each point between n + every - period)]/number of points (rows)
...

Reply:
I added a visual representation instead:

![image](https://user-images.githubusercontent.com/181484/78714631-7090de00-78d9-11ea-86f0-a94a9314ce00.png)

Reply:
excellent, looks great!